test/functional/admin/editions_controller_test.rb

```diff
@@ -309,6 +309,25 @@ class Admin::EditionsControllerTest < ActionController::TestCase
     assert_response :forbidden
   end
 
+  test "passes raw JSON to the export worker" do
+    login_as :gds_editor
+    Admin::EditionFilter.any_instance.stubs(exportable?: true)
+
+    stub = proc do |params, _user_id|
+      assert_not params.instance_of?(ActiveSupport::HashWithIndifferentAccess)
+    end
+
+    DocumentListExportWorker.stub :perform_async, stub do
+      post :export,
```

Comment:
Does this need to be passed as a block, as opposed to just called on the line after the stub is set up? It makes the test a little confusing to read, to my eyes

Reply:
From, a quick go, no it doesn't:

```ruby
    DocumentListExportWorker.stub :perform_async, stub

    post :export,
          params: {
            include_last_author: true,
            include_link_check_reports: true,
            include_unpublishing: true,
            state: "active",
          }
```

...causes:

```

Error:
Admin::EditionsControllerTest#test_passes_raw_JSON_to_the_export_worker:
NoMethodError: undefined method `[]' for nil:NilClass
    test/functional/admin/editions_controller_test.rb:320:in `block in <class:EditionsControllerTest>'
```

Reply:
Fair enough, I don't really understand what's going on there 😅